From 45014e387516949a3a2b1d5862fe85814c6d8c19 Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Thu, 18 Nov 2021 17:59:44 -0800
Subject: Tune TestNamespaceController timeouts (#36137)

Fixes https://github.com/istio/istio/issues/36126

Make timeout for expected case higher as we are hitting timeout in
extreme rare case. Make timeout for unexpected case (IE in success we
wait AT LEAST this long) to make test faster.
---
 .../kube/controller/namespacecontroller_test.go               | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pilot/pkg/serviceregistry/kube/controller/namespacecontroller_test.go b/pilot/pkg/serviceregistry/kube/controller/namespacecontroller_test.go
index a140a599c4..4ce81d9f38 100644
--- a/pilot/pkg/serviceregistry/kube/controller/namespacecontroller_test.go
+++ b/pilot/pkg/serviceregistry/kube/controller/namespacecontroller_test.go
@@ -109,7 +109,7 @@ func expectConfigMap(t *testing.T, client listerv1.ConfigMapLister, ns string, d
 			return fmt.Errorf("data mismatch, expected %+v got %+v", data, cm.Data)
 		}
 		return nil
-	}, retry.Timeout(time.Second*2))
+	}, retry.Timeout(time.Second*10))
 }
 
 func expectConfigMapNotExist(t *testing.T, client listerv1.ConfigMapLister, ns string) {
@@ -117,7 +117,7 @@ func expectConfigMapNotExist(t *testing.T, client listerv1.ConfigMapLister, ns s
 	err := retry.Until(func() bool {
 		_, err := client.ConfigMaps(ns).Get(CACertNamespaceConfigMap)
 		return err == nil
-	}, retry.Timeout(time.Second*2))
+	}, retry.Timeout(time.Millisecond*100))
 
 	if err == nil {
 		t.Fatalf("%s namespace should not have istio-ca-root-cert configmap.", ns)
-- 
2.35.3

